File: python/BOJ/11XXX/11657.py
"""
https://www.acmicpc.net/problem/11657
11657.타임머신
풀이1.1284ms
"""
import sys

input = sys.stdin.readline

n, m = map(int, input().split())
distance = [sys.maxsize] * (n + 1)
distance[1] = 0
is_cycle = False
edges = [tuple(map(int, input().split())) for _ in range(m)]

for _ in range(n - 1):
    for edge in edges:
        (v1, v2, cost) = edge
        if distance[v1] == sys.maxsize:
            continue

        if distance[v1] + cost < distance[v2]:
            distance[v2] = distance[v1] + cost

for edge in edges:
    (v1, v2, cost) = edge
    if distance[v1] == sys.maxsize:
        continue

    if distance[v1] + cost < distance[v2]:
        is_cycle = True
        break

if is_cycle:
    print("-1")
else:
    answer = []
    for d in distance[2:]:
        if d == sys.maxsize:
            answer.append("-1")
        else:
            answer.append(str(d))
    print("\n".join(answer))

File: kotlin/src/main/kotlin/boj/11000/11657.kt
package boj.`11000`

import java.util.StringTokenizer

/**
 * https://www.acmicpc.net/problem/11657
 * 11657.타임머신
 * 풀이1.428ms
 */

data class Edge(val from: Int, val to: Int, val cost: Int)

fun main() {
    val (n, m) = readln().split(" ").map { it.toInt() }
    val edges = mutableListOf<Edge>()
    val distance = LongArray(n + 1) { Long.MAX_VALUE }
    distance[1] = 0

    repeat(m) {
        val stringTokenizer = StringTokenizer(readln())
        edges.add(
            Edge(
                stringTokenizer.nextToken().toInt(),
                stringTokenizer.nextToken().toInt(),
                stringTokenizer.nextToken().toInt()
            )
        )
    }

    repeat(n - 1) {
        for (edge in edges) {
            if (distance[edge.from] == Long.MAX_VALUE)
                continue

            if (distance[edge.from] + edge.cost < distance[edge.to]) {
                distance[edge.to] = distance[edge.from] + edge.cost
            }
        }
    }

    var isCycle = false

    for (edge in edges) {
        if (distance[edge.from] == Long.MAX_VALUE)
            continue

        if (distance[edge.from] + edge.cost < distance[edge.to]) {
            isCycle = true
            break
        }
    }

    if (isCycle)
        print("-1")
    else
        print(distance
            .slice(2 until distance.size)
            .map { if (it == Long.MAX_VALUE) -1 else it }
            .joinToString("\n"))

}